specs/Standards/MultiToken/README.md

```diff
@@ -0,0 +1,539 @@
+### Resolver Trait
+#### Notes
+- TokenId is of type String
+```
+/// Used when MultiTokens are transferred using `mt_transfer_call`. This is the method that's called after `mt_on_transfer`. This trait is implemented on the MultiToken contract.
+pub trait MultiTokenResolver {
+    /// Finalize an `mt_transfer_call` chain of cross-contract calls.
+    ///
+    /// The `mt_transfer_call` process:
+    ///
+    /// 1. Sender calls `mt_transfer_call` on MultiToken contract
+    /// 2. MultiToken contract transfers token from sender to receiver
+    /// 3. MultiToken contract calls `mt_on_transfer` on receiver contract
+    /// 4+. [receiver contract may make other cross-contract calls]
+    /// N. MultiToken contract resolves promise chain with `mt_resolve_transfer`, and may
+    ///    transfer token back to sender
+    ///
+    /// Requirements:
+    /// * Contract MUST forbid calls to this function by any account except self
+    /// * If promise chain failed, contract MUST revert token transfer
+    /// * If promise chain resolves with `true`, contract MUST return token to
+    ///   `sender_id`
+    ///
+    /// Arguments:
+    /// * `previous_owner_id`: the owner prior to the call to `mt_transfer_call`
+    /// * `receiver_id`: the `receiver_id` argument given to `mt_transfer_call`
+    /// * `token_ids`: the `token_ids` argument given to `mt_transfer_call`
+    /// * `approvals`: if using Approval Management, contract MUST provide
+    ///   set of original approved accounts in this argument, and restore these
+    ///   approved accounts in case of revert. In this case it may be multiple sets of approvals
+    ///
+    /// Returns true if tokens were successfully transferred to `receiver_id`.
+    fn mt_resolve_transfer(
+        &mut self,
+        sender_id: AccountId,
+        receiver_id: AccountId,
+        token_ids: Vec<TokenId>,
+        amounts: Vec<U128>,
+    ) -> Vec<U128>;
+}
+```
+### Storage Management Trait
+#### Notes
+This is semi necessary for ft token types to be able to refund users for storage of many different token types like gold/silver... this might be slightly out of scope
+```
+pub trait StorageManagement {
+    // if `registration_only=true` MUST refund above the minimum balance if the account didn't exist and
+    //     refund full deposit if the account exists.
+    fn storage_deposit(
+        &mut self,
+        token_ids: Vec<TokenId>,
+        account_id: Option<AccountId>,
+        registration_only: Option<bool>,
+    ) -> StorageBalance;
+
+    /// Withdraw specified amount of available Ⓝ for predecessor account.
+    ///
+    /// This method is safe to call. It MUST NOT remove data.
+    ///
+    /// `amount` is sent as a string representing an unsigned 128-bit integer. If
+    /// omitted, contract MUST refund full `available` balance. If `amount` exceeds
+    /// predecessor account's available balance, contract MUST panic.
+    ///
+    /// If predecessor account not registered, contract MUST panic.
```

Comment:
`predecessor account` is referenced throughout this, but the arguments are `account_id` -- is `predecessor account` the account from `account_id`?  If so, can we refer to it by its argument name instead? If not, what is the difference between the predecessor account and `account_id`?

Reply:
Predecessor account talks mostly about the calling context of the methods, and account_id is account_id because anyone can make a deposit for any account. 

In the case that I want to pay for your storage on my account, I can call storage_deposit(ids,Some(daryl),None);
The account calling context though it is implementation specific would be env::predecessor::account_id(), which would mean I make a deposit on your behalf. 

In the instance of the other methods, The comments I believe help scope the context in which the methods are intended to be called. So the withdraw is scoped to the predecessor. 

There is no ability within storage management for the payer, to withdraw the deposit they paid on the behalf of the user. This can be done, but in a different scope. So the other thing here is that with force, it would yield power to wipe out or burn holdings of the existing user. It would be a kind of back door of messing someone over through storage management.

That kind of backdoor is also why there's a 1-yocto deposit for withdraw to force a wallet confirmation. 


This is also mostly copy pasta from storage management, with adjustments to deal with batching and token_ids.